0.17/files/create-settings.sh

```diff
@@ -0,0 +1,126 @@
+#! /bin/bash
+
+
+
+
+# Template String
+# Default Value
+# ENvironment Variable String
+updateTemplate () {
+	#get the value from the environment varaible and put it in val
+	val=${!3}
+
+	#Test if there was a value that was set.
+	if [ -z "$val" ]
+	then
+		#replace the value in the template file with the default value
+		sed -i "s/$1/$2/g" "$TEMPLATE_FILE"
+	else
+		#Replace the value in the template file with the value in the environment Variable
+		sed -i "s/$1/$val/g" "$TEMPLATE_FILE"
+	fi
+	
+}
+
+# Template String
+# Default Value
+# ENvironment Variable String
+updateTemplateBool (){
+	#get the value from the environment varaible and put it in val
+	val=${!3}
+	#Test if there was a value that was set to true or false
+	if [[ ${val} =~ ^(true)|(false)$ ]]
+	then
+		#replace the value in the template file with the argument
+		sed -i "s/$1/$val/g" "$TEMPLATE_FILE"
+	else
+		#Replace the value in the template file with the default
+		sed -i "s/$1/$2/g" "$TEMPLATE_FILE"
+	fi
+}
+
+# Template String
+# Default Value
+# ENvironment Variable String
+updateTemplateNumber (){
+	#get the value from the environment varaible and put it in val
+	val=${!3}
+	#Test if there was a value that was set to true or false
+	if [[ ${val} =~ ^[0-9]+$ ]]
+	then
+		#replace the value in the template file with the argument
+		sed -i "s/$1/$val/g" "$TEMPLATE_FILE"
+	else
+		#Replace the value in the template file with the default
+		sed -i "s/$1/$2/g" "$TEMPLATE_FILE"
+	fi
+}
+
+# Template String
+# ENvironment Variable String
+updateTemplateEmptyDefault(){
+	#get the value from the environment varaible and put it in val
+	val=${!2}
+
+	#Test if there was a value that was set.
+	if [ -z "$val" ]
+	then
+		#replace the value in the template file with the default value
+		sed -i "s/$1//g" "$TEMPLATE_FILE"
+	else
+		#Replace the value in the template file with the value in the environment Variable
+		sed -i "s/$1/$val/g" "$TEMPLATE_FILE"
+	fi
+}
+
+mapSettings () {
+	echo Creating map generation settings file
```

Comment:
Don't think this needs to be a function. This can be written at the bottom.

Reply:
The goal of making them functions is twofold. First, it separates the individual file template sections explicitly (I just havent filled the others out yet). Second, because they are called later in the program, they are parsed and validated before being run.

Reply:
Oh, I thought that this is done. Let me know when you filled those in.